js/src/services/vue.service.js

```diff
@@ -79,26 +103,40 @@ class VueService {
    * @param component
    * @param data
    */
-    createVue(name, componentName, component, data) {
-        this.vues.push({
-            name: name,
+    createVue(id, componentName, component, data) {
+        this.registerComponent({
+            ids: [id],
+            name: componentName,
             instance: new Vue({
-                el: name,
-                data: { initData: data },
+                el: id,
+                data: { initData: data, isReady: true },
                 components: { [componentName]: window[component] },
                 mixins: [this.vueMixins],
             }),
+            isLoaded: true,
         });
     }
 
+    /*
+    *  Add component to vues container.
+    *  Group ids that are using the same component.
+     */
+    registerComponent(component) {
+        // check if that component is already registered
+        const registered = this.vues.filter((comp) => comp.name === component.name);
```

Comment:
like we have below `areComponentsLoaded` we can add `isComponentLoaded(name)`

Reply:
sure, I can add it. But I just do not see any good reason why you would check for a specific component only. Do you?

Reply:
Some JS code may need to check if some component is loaded or not, `areComponentsLoaded` can/should use this function plus this check where commetend.

Reply:
I really do not think this is necessary. Since we are talking about Vue components only, the best way to determine if a component is ready or not is via the component itself, from the 'mounted' hook method for example. areComponentsLoaded is really only use for Behat.

Reply:
ok, if needed, we can change later... :)